packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -302,6 +479,34 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
         store.payOracleFeesErc20(address(collateralCurrency), amount);
     }
 
+    function _calculateProposalRewardPct(
+        address perpetual,
+        uint256 startTime,
+        uint256 endTime,
+        FixedPoint.Signed memory proposedRate,
+        FixedPoint.Signed memory currentRate
+    ) private view returns (FixedPoint.Unsigned memory reward) {
+        uint256 timeDiff = endTime.sub(startTime);
+
+        FixedPoint.Unsigned memory rewardRate = _getFundingRateRecord(perpetual).rewardRatePerSecond;
+
+        // First compute the reward for the time elapsed.
+        reward = rewardRate.mul(timeDiff);
+
+        // Next scale the reward based on the absolute difference % between the current and proposed rates.
+        FixedPoint.Signed memory diff = (
+            currentRate.isGreaterThan(proposedRate) ? currentRate.sub(proposedRate) : proposedRate.sub(currentRate)
+        );
```

Comment:
Why are you trying to make the result positive up here if you just handle negatives below anyway (when you compute the absolute value)?

Reply:
good point fixed